src/util/AbortionHandle.cpp

```diff
@@ -0,0 +1,18 @@
+//   Copyright 2023, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#include "AbortionHandle.h"
+
+#include "util/Exception.h"
+
+namespace ad_utility {
+
+void AbortionHandle::abort(AbortionState reason) {
+  AD_CONTRACT_CHECK(reason != AbortionState::NOT_ABORTED);
+
+  static AbortionState NOT_ABORTED = AbortionState::NOT_ABORTED;
+  abortionState_.compare_exchange_strong(NOT_ABORTED, reason,
```

Comment:
But how can it be a race condition if the state is constant?
From https://en.cppreference.com/w/cpp/language/storage_duration
> If multiple threads attempt to initialize the same static local variable concurrently, the initialization occurs exactly once (similar behavior can be obtained for arbitrary functions with [std::call_once](https://en.cppreference.com/w/cpp/thread/call_once)).

> Note: usual implementations of this feature use variants of the double-checked locking pattern, which reduces runtime overhead for already-initialized local statics to a single non-atomic boolean comparison.

But it might be constant-initialization which eliminates this overhead entirely by initializing the variable on application start

Reply:
Nevermind, you're correct